### DISCUSSION TITLE
How to install and use python 3.10+

### DISCUSSION THREAD

**Discussion by umair6** [codemagic-ci-cd/codemagic-docs#2889] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

I am facing issues with installing m2crypto with python 3.8 on apple silicon Mac used by codemagic. Is there any way I can upgraded python to 3.10+ on codemagic's m1,m2 Mac?  

### Expected results

Able to run my script using Python3.10+ 

### Actual results

Unable to install Python3.10+

**Comment by dam-ease** [comment, marked as answer]
Hello @umair6.
Xcode [16.1.x](https://docs.codemagic.io/specs-macos/xcode-16-1/) and [16.2.x](https://docs.codemagic.io/specs-macos/xcode-16-2/) on the mac machines have python `3.12.7` pre-installed. What xcode version are you currently using? Can you please try using any of these?
